### PR TITLE
py sol for matching

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by freakin23** [comment]
AC: https://atcoder.jp/contests/dp/submissions/57707254

File: solutions/gold/ac-matching.mdx [hunk 1]
```diff
@@ -114,4 +114,39 @@ public class Main {
 
 </JavaSection>
 
+<PySection>
+
+```py
+MOD = int(1e9 + 7)
+MAX_N = 21
+
+N = int(input())
```
Comment (SansPapyrus683) [inline solutions/gold/ac-matching.mdx:123]: rename to `n`

File: solutions/gold/ac-matching.mdx [hunk 2]
```diff
@@ -114,4 +114,39 @@ public class Main {
 
 </JavaSection>
 
+<PySection>
+
+```py
+MOD = int(1e9 + 7)
```
Comment (SansPapyrus683) [inline solutions/gold/ac-matching.mdx:120]:
```suggestion
MOD = 10 ** 9 + 7
```

File: solutions/gold/ac-matching.mdx [hunk 3]
```diff
@@ -114,4 +114,39 @@ public class Main {
...
+MAX_N = 21
```
Comment (SansPapyrus683) [inline solutions/gold/ac-matching.mdx:121]: 21? or just 20?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/ac-matching.mdx (hunk 3 above)
Comment (freakin23) [reply solutions/gold/ac-matching.mdx:121]: It was 21 in C++ version

File: solutions/gold/ac-matching.mdx (hunk 1 above)
Comment (freakin23) [reply solutions/gold/ac-matching.mdx]: done

**Comment by freakin23** [comment]
AC: https://atcoder.jp/contests/dp/submissions/57717496

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**